tracer/src/Datadog.Trace/ContinuousProfiler/ProfilerStatus.cs

```diff
@@ -5,48 +5,30 @@
 
 using System;
 using System.Runtime.InteropServices;
-using Datadog.Trace.ExtensionMethods;
 using Datadog.Trace.Logging;
-using Datadog.Trace.Util;
 
 namespace Datadog.Trace.ContinuousProfiler
 {
     internal class ProfilerStatus : IProfilerStatus
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProfilerStatus));
 
-        private readonly bool _isProfilingEnabled;
+        private readonly ProfilerState _profilerState;
         private readonly object _lockObj;
         private bool _isInitialized;
         private IntPtr _engineStatusPtr;
 
-        public ProfilerStatus()
+        public ProfilerStatus(ProfilerSettings settings)
         {
-            var fd = FrameworkDescription.Instance;
-            var isSupported =
-                (fd.OSPlatform == OSPlatformName.Windows && (fd.ProcessArchitecture == ProcessArchitecture.X64 || fd.ProcessArchitecture == ProcessArchitecture.X86)) ||
-                (fd.OSPlatform == OSPlatformName.Linux && fd.ProcessArchitecture == ProcessArchitecture.X64);
-
-            _isProfilingEnabled = false;
-
-            if (isSupported)
+            _profilerState = settings.ProfilerState;
+            var state = _profilerState switch
             {
-                var manualDeployement = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.ProfilingEnabled);
-                if (manualDeployement != null)
-                {
-                    // it is possible that SSI installation script is setting the environment variable to "auto" to enable the profiler
-                    // instead of "true" to avoid starting the profiler immediately after the installation
-                    _isProfilingEnabled = manualDeployement.ToBoolean() ?? (manualDeployement == "auto");
-                }
-                else
-                {
-                    // the profiler is declared "enabled" just if the SSI environment variable exists to be sure that telemetry metrics
-                    // will contain the right status (i.e. we need the tracer to send the spans even if the profiler is not started yet)
-                    _isProfilingEnabled = (EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.SsiDeployed) != null);
-                }
-            }
+                ProfilerState.Enabled => "enabled",
+                ProfilerState.Auto => "auto",
+                _ => "disabled"
+            };
 
-            Log.Information("Continuous Profiler is {IsEnabled}.", _isProfilingEnabled ? "enabled" : "disabled");
+            Log.Information("Continuous Profiler mode = {ProfilerState}", state);
```

Comment:
Maybe I'm not used to the pattern used to log in the tracer, but why do you add {ProfilerState} in the log as a string and not an interpolated string?

I'm expecting more something like:
`   Log.Information("Continuous Profiler mode = ", state);`
or 
`   Log.Information($"Continuous Profiler mode = {state}");`

Reply:
In general, you should never use interpolated strings for logging. We even have analyzers that warns you not to 😄 There's three reasons for that:

- It breaks the caching that the log framework does related to parsing out parameters
- It means you don't get structured logging (not actually an issue for us, we log to a flat file, but 'good practice' in general)
- For error logs, we explicitly are only allowed to log "constant" values for security reasons, and interpolation doesn't satisfy that (unless all the parameters are _also_ constants)

In this case, it actually _would_ be fine, because the log is only written once and it's not an error, but I didn't think it was worth creating an explicit `#pragma` to allow it 😄

Reply:
I get it but why {ProfilerState} as "{ProfilerState}" immediately concatenated to the value of `state`?
We'll get something like the following in the logs:
`   Continuous Profiler = {ProfilerState}Enabled`

I would expect instead:
`   Continuous Profiler mode = Enabled`

Reply:
Think of it as somewhat equivalent to

```csharp
Log.Information(string.Format("Continuous Profiler mode = {0}", state));
```

but where you _also_ have an associated `ProfilerState=state` ambient variable 🙂

Reply:
Sorry... My mind is so twisted by C++   ;^)